common/src/main/java/com/graphicsfuzz/common/util/FragmentShaderToShaderJob.java

```diff
@@ -36,27 +61,133 @@ private FragmentShaderToShaderJob() {
+    // randomized value (using 'generator' as the source of randomness).
+    for (VariablesDeclaration vd : tu.getUniformDecls()) {
+      if (!(vd.getBaseType().getWithoutQualifiers() instanceof BasicType)) {
+        throw new RuntimeException("Non-basic types not implemented, found "
+            + vd.getBaseType().toString());
+      }
+      BasicType basicType = (BasicType) vd.getBaseType().getWithoutQualifiers();
+      for (VariableDeclInfo vdi : vd.getDeclInfos()) {
+        int arrayLength;
+        if (vdi.hasArrayInfo()) {
+          arrayLength = vdi.getArrayInfo().getConstantSize();
+        } else {
+          arrayLength = 1;
+        }
+        List<Number> values = new ArrayList<>();
+        for (int i = 0; i < basicType.getNumElements() * arrayLength; i++) {
+          if (basicType.getElementType() == BasicType.FLOAT) {
+            values.add(generator.nextFloat());
+          } else if (basicType.getElementType() == BasicType.INT) {
+            values.add(generator.nextInt(0xffffff));
+          } else if (basicType.getElementType() == BasicType.UINT) {
+            values.add(generator.nextPositiveInt(0xffffff));
+          } else {
+            assert basicType.getElementType() == BasicType.BOOL;
+            values.add(generator.nextBoolean() ? 1 : 0);
+          }
+        }
+        pipelineInfo.addUniform(vdi.getName(), basicType, maybeGetArrayCount(vdi), values);
+      }
+    }
 
-    // TODO(https://github.com/google/graphicsfuzz/issues/837): If the translation unit uses 'in'
-    //  global variables, declare a vertex shader TranslationUnit with corresponding 'out' variables
-    //  and add it to 'shaders'.
 
+    // If the translation unit uses 'in' global variables, declare a vertex shader
+    // TranslationUnit with corresponding 'out' variables and add it to 'shaders'.
+    boolean hasInputs = false;
+    TranslationUnit vertexShader = null;
+
+    for (VariablesDeclaration vd : tu.getGlobalVariablesDeclarations()) {
+      if (vd.getBaseType().hasQualifier(TypeQualifier.SHADER_INPUT)) {
+        BasicType basicType = (BasicType) vd.getBaseType().getWithoutQualifiers();
+        // Create clone of the variables declaration and change the input to output:
+        QualifiedType varType = (QualifiedType) vd.getBaseType().clone();
+        varType.replaceQualifier(TypeQualifier.SHADER_INPUT, TypeQualifier.SHADER_OUTPUT);
+        // Check if this is the first input we're handling
+        if (!hasInputs) {
+          // Since we're going to need it, initialize vertexShader with
+          // skeleton definition
+          vertexShader = new TranslationUnit(ShaderKind.VERTEX,
+              Optional.of(ShadingLanguageVersion.ESSL_310), Arrays.asList(
+              new PrecisionDeclaration("precision mediump float;"),
+              new FunctionDefinition(
+                  new FunctionPrototype("main", VoidType.VOID, Collections.emptyList()),
```

Comment:
Maybe use Collections.emptyList() again, instead of new ArrayList<>().

Reply:
Done (brainfart)